### DISCUSSION TITLE
ESP32-MINI-1 support

### DISCUSSION THREAD

**Discussion by ThomasNygards** [micropython/micropython#10344] [ESP32] [answered]
Hi.
I have aESP32-DevKitM-1 card which is ESP32-MINI-1(1U) based.
[see datasheet](https://www.espressif.com/sites/default/files/documentation/esp32-mini-1_datasheet_en.pdf)

I have flashed it successfully with this firmware
https://micropython.org/download/esp32/
But connecting to it using Thonny I get this message:

`ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 188777542, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:4540
ho 0 tail 12 room 4
load:0x40078000,len:12448
load:0x40080400,len:4124
entry 0x40080680
**E (624) cpu_start: Running on single core chip, but application is built with dual core support.
E (624) cpu_start: Please enable CONFIG_FREERTOS_UNICORE option in menuconfig.**
Re-enable cpu cache.

abort() was called at PC 0x40081bf9 on core 0

Backtrace:0x40093b5e:0x3ffe3bc0 0x400942d5:0x3ffe3be0 0x40097f26:0x3ffe3c00 0x40081bf9:0x3ffe3c70 0x40078f60:0x3ffe3c90 |<-CORRUPTED

ELF file SHA256: 00d24fd4330e05f8

Rebooting...`

Is there no support for this ESP32 for MicroPython or how do I solve this?
I have seen this post from [AwesomeCronk](https://forum.micropython.org/memberlist.php?mode=viewprofile&u=5893):
https://forum.micropython.org/viewtopic.php?f=18&t=12022
But how do I use that?

**Comment by robert-hh** [comment]
See https://forum.micropython.org/viewtopic.php?f=18&t=12022&p=65604&hilit=unicore#p65604 and https://github.com/micropython/micropython/pull/8365
It requires a specific board configuration for which there is a PR, which was reviewed but it was never merged. Maybe because of an error in the commit message format, which was not fixed.

**Comment by robert-hh** [comment, marked as answer]
> But how do I use that?

You have to set up the build environment and build the firmware using this PR. That's quite s bit of work, if doing it the first time. Instructions are in the readme file for the esp32 port and for MicroPython.
